guiqwt/plot.py

```diff
@@ -1248,8 +1248,6 @@ def __init__(
         self.xcsw = XCrossSection(self)
         self.xcsw.setVisible(show_xsection)
 
-        self.xcsw.SIG_VISIBILITY_CHANGED.connect(self.xcsw_is_visible)

```

Comment:
Could you confirm that, if this signal stays connected, it does not have any impact on your issue?
(this signal connection is really important)

Reply:
In other words, before merging, this PR, I would prefer to keep this signal connection and simply remove the QApplication.processEvents call.

Reply:
> Could you confirm that, if this signal stays connected, it does not have any impact on your issue? (this signal connection is really important)

yes if the signal stays connected and I remove the calls to processEvents() it works 100% and I do not see any behaviour issues